e2e/gui/src/test/java/com/epam/pipeline/autotests/RestrictionsOnInstancePriceTypeTest.java

```diff
@@ -93,74 +110,562 @@ public void deletingEntities() {
+        try {
+            String[] masks = clusterAllowedMasks.split(",");
+            loginAs(admin);
+            setClusterAllowedStringPreference(clusterAllowedInstanceTypes, format("%s*", masks[0]));
+            setClusterAllowedStringPreference(clusterAllowedInstanceTypesDocker, format("%s*", masks[1]));
+            setMaskForUser(user.login, instanceTypesMask, format("%s*", masks[2]));
+            setMaskForUser(user.login, toolInstanceTypesMask, format("%s*", masks[3]));
+            tools()
+                    .performWithin(defaultRegistry, defaultGroup, testingTool, tool ->
+                            tool.showInstanceManagement(instanceManagement ->
+                                    instanceManagement
+                                            .addAllowedToolInstanceTypesMask(format("%s.*", instanceFamilyName))
+                                            .clickApply()
+                                            .sleep(2, SECONDS)));
+            tools()
+                    .perform(defaultRegistry, defaultGroup, testingTool, ToolTab::runWithCustomSettings)
+                    .expandTab(EXEC_ENVIRONMENT)
+                    .checkValueIsInDropDown(INSTANCE_TYPE, instanceFamilyName);
+            logout();
+            loginAs(user);
+            tools()
+                    .perform(defaultRegistry, defaultGroup, testingTool, ToolTab::runWithCustomSettings)
+                    .expandTab(EXEC_ENVIRONMENT)
+                    .checkValueIsInDropDown(INSTANCE_TYPE, masks[3]);
+        } finally {
+            logout();
+            loginAs(admin);
+            setClusterAllowedStringPreference(clusterAllowedInstanceTypes, defaultClusterAllowedInstanceTypes);
+            setClusterAllowedStringPreference(clusterAllowedInstanceTypesDocker, defaultClusterAllowedInstanceTypes);
+            setMaskForUser(user.login, instanceTypesMask, "");
+            setMaskForUser(user.login, toolInstanceTypesMask, "");
+            tools()
+                    .performWithin(defaultRegistry, defaultGroup, testingTool, tool ->
+                            tool.showInstanceManagement(instanceManagement ->
+                                    instanceManagement
+                                            .addAllowedToolInstanceTypesMask("")
+                                            .clickApply()
+                                            .sleep(2, SECONDS)));
+        }
+    }
+
+    @CloudProviderOnly(values={Cloud.AWS,Cloud.GCP})
+    @Test(dependsOnMethods={"preparationForValidationOfInstanceTypesRestrictions"})
+    @TestCase({"EPMCMBIBPC-2648"})
+    public void validationOfPriceTypesRestrictionsOverUserManagement() {
+        try {
+            loginAs(admin);
+            openEditUserTab(user.login)
+                    .setAllowedPriceType(onDemandPrice)
+                    .ok();
+            logout();
+            loginAs(user);
+            validationOfPriceTypesRestrictions("On-demand", "On-demand", "On-demand");
```

Comment:
We have Utils.ON_DEMAND constant, let's replace in all places of the class

Reply:
done